### ISSUE TITLE
Focus trap / inert mixin

### ISSUE THREAD

**Issue by smhigley** [dojo/widget-core#730]
**Enhancement**

From the widgets focus management issue (https://github.com/dojo/widgets/issues/309#issuecomment-339100993):

> **Either trap focus within a node, or remove it from the focus order/AOM**
>
> Both these functions rely on inert, which I think we should use and polyfill. We can also either include blockingElements and its polyfill, or create our own implementation. I'm a little dubious about blockingElements actually becoming spec, at least in the foreseeable future. On the other hand, using the polyfill would mean we don't need to traverse the DOM ourselves to apply inert (this is essentially what the polyfill does).
>
> I believe this would also make sense as a mixin that adds `inert` and `blocking` to the properties interface.

I'm not sure what this should be called, since its use isn't limited to focus traps. Perhaps `FocusOrderMixin`?
